public/app/features/scopes/dashboards/ScopesDashboardsService.ts

```diff
@@ -81,40 +72,28 @@ export class ScopesDashboardsService extends ScopesServiceBase<ScopesDashboardsS
       return;
     }
 
-    this._fetchSub?.unsubscribe();
-
     if (forScopeNames.length === 0) {
       this.updateState({
         dashboards: [],
         filteredFolders: {},
         folders: {},
         forScopeNames: [],
         loading: false,
+        drawerOpened: false,
       });
 
-      ScopesService.instance?.setDrawerOpened(false);
-
       return;
     }
 
     this.updateState({ forScopeNames, loading: true });
 
-    this._fetchSub = from(this.fetchDashboardsApi(forScopeNames))
-      .pipe(
-        finalize(() => {
-          this.updateState({ loading: false });
-        })
-      )
-      .subscribe((dashboards) => {
-        const folders = this.groupDashboards(dashboards);
-        const filteredFolders = this.filterFolders(folders, this.state.searchQuery);
-
-        this.updateState({ dashboards, filteredFolders, folders, loading: false });
-
-        ScopesService.instance?.setDrawerOpened(dashboards.length > 0);
+    const dashboards = await this.apiClient.fetchDashboards(forScopeNames);
+    if (isEqual(this.state.forScopeNames, forScopeNames)) {
```

Comment:
Instead of the observable wrapper for cancelation we just check if the thing we fetched is the same as in the `state.forScopeNames` if not we discard it.